### PR TITLE
Federated unfollow

### PR DESCRIPTION
**Pull request by dessalines**

### CODE REVIEW

**Comment by Nutomic** [comment]
This works really nicely! One problem I noticed is that only the instance that deleted a post/comment can restore it again. On the other instance it shows  the post/comment as deleted, but still shows a delete button (which does nothing).

File: server/src/api/community.rs [hunk 1]
```diff
@@ -495,9 +501,19 @@ impl Perform for Oper<FollowCommunity> {
...
       let user = User_::read(&conn, user_id)?;
-      user.send_follow(&community.actor_id, &conn)?;
+
+      if data.follow {
+        // Dont actually add to the community followers here, because you need
+        // to wait for the accept
+        user.send_follow(&community.actor_id, &conn)?;
+      } else {
+        user.send_unfollow(&community.actor_id, &conn)?;
+        match CommunityFollower::ignore(&conn, &community_follower_form) {
```
Comment (Nutomic) [inline server/src/api/community.rs:512]: Shouldnt this be `unfollow()`? Ignore is confusing.

Comment (dessalines) [reply server/src/api/community.rs:512]: Agree, this is an easy enough change since its in the trait.

**Comment by dessalines** [comment]
> This works really nicely! One problem I noticed is that only the instance that deleted a post/comment can restore it again. On the other instance it shows the post/comment as deleted, but still shows a delete button (which does nothing).

I'm trying this out now.

**Comment by dessalines** [comment]
> 

    This works really nicely! One problem I noticed is that only the instance that deleted a post/comment can restore it again. On the other instance it shows the post/comment as deleted, but still shows a delete button (which does nothing).

This is for remove right. This might just have to do with not being able to mod your own comments.

**Pull request merged**